### PR TITLE
[FUSE-13] Implements CRUD for User Service

### PR DESCRIPTION
**Pull request by CaoSon1909**
Implements CRUD for User management, includes: Admin, Customer, Trainer

### CODE REVIEW

File: apps/weight-loss-consultant-users-mgnt-api/src/app/controllers/admin.controller.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    return this.userService.findAll();
+  }
+
+  @Get(':email')
+  @ApiResponse({ status: 200, description: 'Admin details has shown below:'})
+  @ApiResponse({ status: 403, description: 'Forbidden.'})
+  @ApiResponse({ status: 404, description: 'Email not found'})
+  @ApiParam({
+    name :  "email",
+    type : "String",
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/controllers/admin.controller.ts:32]: This should be type String not "String"

File: apps/weight-loss-consultant-users-mgnt-api/src/app/controllers/admin.controller.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    required : true
+  }
+)
+  async getByEmail(@Param('email') email: string, @Res() res): Promise<any> {
+    try {
+      const admin = await this.userService.viewDetail(email);
+      res.status(200).send(admin);
+    }
+    catch (e) {
+      console.log(e);
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/controllers/admin.controller.ts:43]: should use console.err(e)

File: apps/weight-loss-consultant-users-mgnt-api/src/app/controllers/customer.controller.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  async getByEmail(@Param('email') email: string): Promise<AdminEntity> {
+    const customer = await this.customerService.viewDetail(email);
+    if (!customer) {
+      throw new NotFoundException(`Not found trainer with email: ${email}`)
+    }
+    return customer;
+  }
+
+  @Post()
+  // @UsePipes(ValidationPipe)
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/controllers/customer.controller.ts:54]: Why not delete this comment?

File: apps/weight-loss-consultant-users-mgnt-api/src/app/dtos/customer/update-customer-dto.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  @IsString()
+  @MinLength(LIMIT_LENGTH.EMAIL_MIN_LENGTH, {message: ERR_MSG.EMAIL_FORMAT_ERR})
+  @MaxLength(LIMIT_LENGTH.EMAIL_MAX_LENGTH, {message: ERR_MSG.EMAIL_MAX_LENGTH_ERR})
+  @ApiProperty({
+    description: 'Email of customer. cannot update',
+    minimum: 1,
+    type: String,
+
+  })
+  email?: string;
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/dtos/customer/update-customer-dto.ts:16]: this property should be added with decorator @IsEmail()

File: apps/weight-loss-consultant-users-mgnt-api/src/app/dtos/match.decorator.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import {
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/dtos/match.decorator.ts:1]: Please give the URL of the Stackoverflow that you copy that from

File: apps/weight-loss-consultant-users-mgnt-api/src/app/dtos/trainer/create-trainer.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+export class CreateTrainerDto{
+  @Validator.IsString({message: ERR_MSG.EMAIL_FORMAT_ERR})
+  @Validator.MinLength(LIMIT_LENGTH.EMAIL_MIN_LENGTH)
+  @Validator.MaxLength(LIMIT_LENGTH.EMAIL_MAX_LENGTH, {message: ERR_MSG.EMAIL_MAX_LENGTH_ERR})
+  @ApiProperty({
+    description: 'Email of trainer, unique',
+    minimum: 1,
+    type: String,
+  })
+  email: string;
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/dtos/trainer/create-trainer.ts:19]: this property should be added with decorator @IsEmail()

File: apps/weight-loss-consultant-users-mgnt-api/src/app/dtos/trainer/update-trainer.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+export class UpdateTrainerDto {
+  @IsString()
+  @MinLength(LIMIT_LENGTH.EMAIL_MIN_LENGTH, {message: ERR_MSG.EMAIL_FORMAT_ERR})
+  @MaxLength(LIMIT_LENGTH.EMAIL_MAX_LENGTH, {message: ERR_MSG.EMAIL_MAX_LENGTH_ERR})
+  @ApiProperty({
+    description: 'Email of trainer. cannot update',
+    minimum: 1,
+    type: String,
+  })
+  email: string;
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/dtos/trainer/update-trainer.ts:15]: this property should be added with decorator @IsEmail()

File: apps/weight-loss-consultant-users-mgnt-api/src/app/mappers/customer.mapper.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    entity.email = dto.email;
+    entity.password = dto.password;
+    entity.fullname = "";
+    entity.address = "";
+    entity.phone = "";
+    entity.gender = "";
+    entity.status = 1;
+    entity.profileImage = "";
+    entity.dob = 1;
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/mappers/customer.mapper.ts:22]: What is this? Why dob is 1?

File: apps/weight-loss-consultant-users-mgnt-api/src/app/mappers/trainer.mapper.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    entity.status = 1;
+    entity.fullname = "";
+    entity.address = "";
+    entity.phone = "";
+    entity.gender = "";
+    entity.profileImage = "";
+    entity.dob = 1;
+    entity.yearOfExp = 1;
+    entity.rating = 1;
+    return entity;
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/mappers/trainer.mapper.ts:26]: Why assign 1 for these properties?

File: apps/weight-loss-consultant-users-mgnt-api/src/app/repositories/admin.repository.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import {EntityRepository, Repository} from "typeorm";
+import {AdminEntity} from "../entities/admin.entity";
+
+
+@EntityRepository(AdminEntity)
+export class AdminRepository extends Repository<AdminEntity> {
+  getAllAdmins(){
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/repositories/admin.repository.ts:7]: Please assign a return type for this method

File: apps/weight-loss-consultant-users-mgnt-api/src/app/services/app.service.ts [hunk 11]
```diff
@@ -3,6 +3,6 @@ import { Injectable } from '@nestjs/common';
 @Injectable()
 export class AppService {
   getData(): { message: string } {
-    return { message: 'Welcome to weight-loss-consultant-users-mgnt-api!' };
+    return { message: 'Welcome to weight-loss-consultant!' };
```
Comment (TienTruong209) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/services/app.service.ts:6]: Please remove this service

**Review by TienTruong209 — CHANGES_REQUESTED** [review]
Thank you Son, but please change what I commented.

File: apps/weight-loss-consultant-authentication/src/constant.ts [hunk 12]
```diff
@@ -1,5 +1,22 @@
...
+export const ormConfig: MysqlConnectionOptions = {
+  type: 'mysql',
+  host: 'ec2-54-251-131-46.ap-southeast-1.compute.amazonaws.com',
+  port: 3306,
+  username: 'root',
+  password: 'Nhatrang@1',
+  database: 'weight_loss_consultant',
+  entities: [TrainerEntity, CustomerEntity]
+};
+
```
[the remark this answers is not in the document]
Comment (CaoSon1909) [reply apps/weight-loss-consultant-authentication/src/constant.ts:14]: Resolved
image

**Pull request merged**